app/assets/javascripts/actions/training_actions.js

```diff
@@ -2,12 +2,67 @@ import { extend } from 'lodash-es';
 import {
   RECEIVE_TRAINING_MODULE, MENU_TOGGLE, REVIEW_ANSWER,
   SET_CURRENT_SLIDE, RECEIVE_ALL_TRAINING_MODULES,
-  EXERCISE_COMPLETION_UPDATE, SLIDE_COMPLETED, API_FAIL
+  EXERCISE_COMPLETION_UPDATE, SLIDE_COMPLETED, API_FAIL,
+  RECEIVE_TRAINING_LIBRARIES, RECEIVE_TRAINING_LIBRARY, SEARCH_LIBRARY
 } from '../constants';
 import request from '../utils/request';
 import logErrorMessage from '../utils/log_error_message';
 import { stringify } from 'query-string';
 
+export const fetchTrainingLibraries = () => async (dispatch) => {
+  try {
+    const response = await request('/training.json');
+    if (!response.ok) {
+      const data = await response.text();
+      response.responseText = data;
+      throw response;
+    }
+    const data = await response.json();
```

Comment:
this can be extracted to a function — its used in a few places even before this PR

Reply:
Hi @TheTrio , sure I can do that.